From e56ce1b1ccb69103c14b74c19a788dbae7c332a5 Mon Sep 17 00:00:00 2001
From: Nick Kralevich <nnk@google.com>
Date: Sat, 19 Nov 2016 12:09:33 -0800
Subject: [PATCH] change /data/bugreports to /bugreports

It's 5 characters shorter, has no runtime costs, can be stored on a
read-only partition, and avoids problems like b/27262109 and b/27204904.
It allows makes some security hardening easier.

(cherry picked from commit c4084c669495fd1b41ce8b6568c71b1f46d26969)

Bug: 27262109
Bug: 27204904
Bug: 32799236
Test: verified new symlink created and old one not present
Change-Id: Ief362e13569ad9c868a7f0f9c4dbd6a328c96c6b
(cherry picked from commit 04bbd5c6dd0455af43e96efbc45399bf6fd70d40)
---
 rootdir/Android.mk | 1 +
 rootdir/init.rc    | 4 ----
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/rootdir/Android.mk b/rootdir/Android.mk
index e48965d..b150001 100644
--- a/rootdir/Android.mk
+++ b/rootdir/Android.mk
@@ -123,6 +123,7 @@ endif
 LOCAL_POST_INSTALL_CMD := mkdir -p $(addprefix $(TARGET_ROOT_OUT)/, \
     sbin dev proc sys system data oem acct config storage mnt root $(BOARD_ROOT_EXTRA_FOLDERS)); \
     ln -sf /system/etc $(TARGET_ROOT_OUT)/etc; \
+    ln -sf /data/user_de/0/com.android.shell/files/bugreports $(TARGET_ROOT_OUT)/bugreports; \
     ln -sf /sys/kernel/debug $(TARGET_ROOT_OUT)/d; \
     ln -sf /storage/self/primary $(TARGET_ROOT_OUT)/sdcard
 ifdef BOARD_USES_VENDORIMAGE
diff --git a/rootdir/init.rc b/rootdir/init.rc
index 9e9f87a..5d7b3f6 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -451,10 +451,6 @@ on post-fs-data
 
     mkdir /data/anr 0775 system system
 
-    # symlink to bugreport storage location
-    rm /data/bugreports
-    symlink /data/user_de/0/com.android.shell/files/bugreports /data/bugreports
-
     # Separate location for storing security policy files on data
     mkdir /data/security 0711 system system
 
-- 
2.7.4

